### PR TITLE
add ngraph::pass::LSTMCellDecomposition as mandatory

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
 - add ngraph::pass::LSTMCellDecomposition as mandatory

### Tickets:
 - 62238

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_plugin.cpp [hunk 1]
```diff
@@ -711,6 +712,7 @@ void GNAPlugin::LoadNetwork(CNNNetwork & _network) {
         manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
         manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
         manager.register_pass<ngraph::pass::ConvertOpSet1ToLegacy>();
+        manager.register_pass<ngraph::pass::LSTMCellDecomposition>();
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:715]: Does LSTMCellDecomposition produce graph in legacy opset? Shouldn't we run it before transformations which covert graph from the latest to the legacy opset?

Comment (nkogteva) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:715]: Maybe there is a sense to put legacy transformations under if condition and don't call them for graph based functions where we have a similar ngraph pass.

File: inference-engine/src/gna_plugin/gna_plugin.cpp [hunk 2]
```diff
@@ -754,12 +760,13 @@ void GNAPlugin::LoadNetwork(CNNNetwork & _network) {
 
     // network optimisation phases
     int passIdx = 0;
-    auto run_passes = [&] (const CNNNetwork& network, bool runBeforeCopy, bool lowPrecision) {
+    auto run_passes = [&] (const CNNNetwork& network, bool runBeforeCopy, bool lowPrecision, bool isNgraphPassesUsed) {
```
Comment (nkogteva) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:763]: Too complicated. `isNgraphPassesUsed` is available here without extending arguments list of the function.

File: inference-engine/src/gna_plugin/frontend/model_quantizer.hpp [hunk 3]
```diff
@@ -26,8 +26,10 @@ namespace GNAPluginNS {
 template<class T>
 class ModelQuantizer {
  public:
+    ModelQuantizer(bool isNgraphPassesUsed) : isNgraphPassesUsed_(isNgraphPassesUsed) {}
```
Comment (nkogteva) [inline inference-engine/src/gna_plugin/frontend/model_quantizer.hpp:29]: Please remove it from quantizer

**Review by nkogteva — APPROVED** [review]

**Review by elilobanova — APPROVED** [review]

**Pull request merged**